examples/src/main/java/io/opencensus/examples/grpc/helloworld/HelloWorldServer.java

```diff
@@ -0,0 +1,189 @@
+      span.setStatus(Status.UNKNOWN);
+    }
+  }
+
+  private void start() throws IOException {
+    server = ServerBuilder.forPort(serverPort)
+        .addService(new GreeterImpl())
+        .build()
+        .start();
+    logger.info("Server started, listening on " + serverPort);
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      @Override
+      public void run() {
+        // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+        System.err.println("*** shutting down gRPC server since JVM is shutting down");
+        HelloWorldServer.this.stop();
+        System.err.println("*** server shut down");
+      }
+    });
+  }
+
+  private void stop() {
+    if (server != null) {
+      server.shutdown();
+    }
+  }
+
+  /**
+   * Await termination on the main thread since the grpc library uses daemon threads.
+   */
+  private void blockUntilShutdown() throws InterruptedException {
+    if (server != null) {
+      server.awaitTermination();
+    }
+  }
+
+  /**
+   * Main launches the server from the command line.
+   */
+  public static void main(String[] args) throws IOException, InterruptedException {
+    int serverPort = getPortOrDefaultFromArgs(args, 0, 50051);
+    String cloudProjectId = getStringOrDefaultFromArgs(args, 1, null);
+    int zPagePort = getPortOrDefaultFromArgs(args, 2, 3000);
+    int prometheusPort = getPortOrDefaultFromArgs(args, 3, 9090);
+
+    // Registers all RPC views.
+    RpcViews.registerAllViews();
+
+    // Registers logging trace exporter.
+    LoggingTraceExporter.register();
+
+    // Starts a HTTP server and registers all Zpages to it.
+    ZPageHandlers.startHttpServerAndRegisterAll(zPagePort);
+    logger.info("ZPages server starts at localhost:" + zPagePort);
+
+    // Registers Stackdriver exporters.
+    if (cloudProjectId != null) {
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder().setProjectId(cloudProjectId).build());
+      StackdriverStatsExporter.createAndRegister(
+          StackdriverStatsConfiguration
+              .builder()
+              .setProjectId(cloudProjectId)
+              .setExportInterval(Duration.create(15, 0))
+              .build());
+    }
+
+    // Register Prometheus exporters and export metrics to a Prometheus HTTPServer.
+    PrometheusStatsCollector.createAndRegister();
+    HTTPServer prometheusServer = new HTTPServer(prometheusPort, true);
+
+    // Start the RPC server. You shouldn't see any output from gRPC before this.
+    logger.info("gRPC starting.");
+    final HelloWorldServer server = new HelloWorldServer(serverPort);
+    server.start();
+    server.blockUntilShutdown();
+  }
+
+  static class GreeterImpl extends GreeterGrpc.GreeterImplBase {
+
+    @Override
+    public void sayHello(HelloRequest req, StreamObserver<HelloReply> responseObserver) {
+      // Trace all incoming RPCs.
+      Tracing.getTraceConfig().updateActiveTraceParams(
```

Comment:
I am not quite sure if we should do this in a multi-threaded RPC handler. Our current implementation of `getTraceConfig()` returns the [global, not thread-local configuration](https://github.com/census-instrumentation/opencensus-java/blob/ee1fd0259f46e21b1a90100ed35926a528d20ee0/impl_core/src/main/java/io/opencensus/implcore/trace/config/TraceConfigImpl.java#L23). 

The only way I can think of to make it sampled now is to invoke `setRecordEvents()` or pre-register the span for collection. @sebright and @bogdandrutu for confirmation.

Reply:
I agree, the RPC shouldn't enable 100% sampling globally.  SpanBuilder allows setting a sampler for the new span, though.

Reply:
Sorry I made a mistake in previous comment. They are for local span collection, not for sampling. We can call `SpanBuilder.setSampler()` to apply the sampler to the span.

Reply:
Done, though I've seen cpp is setting global trace params: https://github.com/census-instrumentation/opencensus-cpp/blob/master/examples/grpc/hello_client.cc#L55.